### ISSUE TITLE
[Feature Request]: Script Access

### ISSUE THREAD

**Issue by JacobPersi** [zero01101/openOutpaint#176]
### Is your feature request related to a problem? Please describe.

I have [a script](https://github.com/devilismyfriend/PXL8) that downscales and reduces image color pallet to make the output image appear like pixel art. I'd love to be able to use this in out-painting, but right now there is no access to the scripts tab.

### Describe the solution you'd like

Some way to run scripts following generation. 

### Proposed workflow

1. Scripts tab added to generation overlay. 

### Describe alternatives you've considered

N/A

**Comment by zero01101** [comment]
moved issue from extension to base repo, adding #duplicate (see https://github.com/zero01101/openOutpaint/issues/163, https://github.com/zero01101/openOutpaint/issues/167)

:/ a few people have requested scripts, but currently the only API access to them is to merely specify a script name string and dictionary of script parameters - there's no way currently to query a list of scripts and their params through the API, and even though i guarantee there's _ways_ to get the scripts from gradio directly, the main repo is its own standalone app and thus kinda has to have feature parity with the extension, other than the gradio-specific send-to buttons... 

for what it's worth though, hey, pxl8's really neat, that's a slick model and using a custom dithering script is super clever - out of curiosity, what's the deal with the .exe?

**Comment by JacobPersi** [comment]
> tl;dr, kinda need an API endpoint to GET scripts/parameters
Is it possible to expose these fields behind some advanced settings? It would be useful for those who want to run scripts even though the UI isn't there.  

> for what it's worth though, hey, pxl8's really neat, that's a slick model and using a custom dithering script is super clever - out of curiosity, what's the deal with the .exe?
I'm not affiliated, but its a compiled version of [nQuantCpp](https://github.com/mcychan/nQuantCpp), they should probably just link to their release though...

**Comment by zero01101** [comment]
gotta admit i hadn't considered like a secret bare text input panel that'd accept a script name and a JSON object of parameters; can't promise anything, but if that ends up happening i'll update here :)

**Comment by zero01101** [comment]
well it's... in progress? https://github.com/zero01101/openOutpaint/issues/163#issuecomment-1399110516

**Comment by zero01101** [comment]
"implemented" in https://github.com/zero01101/openOutpaint/commit/c0e936f2166e8d055a88a30911d57d7094a79a74, a little janky but [the manual](https://github.com/zero01101/openOutpaint/wiki/Manual#script) should give you a good start :)
